src/components/Overlay/Overlay.component.tsx

```diff
@@ -0,0 +1,54 @@
+import React, {useRef, useEffect} from 'react';
+
+import {OverlayProps} from './Overlay';
+import {OverlayContainer, StyledPopover, EventPassContainer} from './Overlay.styles';
+
+export const Overlay = ({
+    show,
+    popoverBody,
+    children,
+    placement = 'top',
+    elevation = false,
+    className = '',
+    setShow,
+}: OverlayProps) => {
+    const EventPassContainerRef = useRef<HTMLDivElement>(null);
+
+    /**
+	 * Function which closes the popover on clicking anywhere outside the popover wrapper
+	 * @param {MouseEvent} e - gets the element on which the onClick event is acted upon
+	 */
+    const onPopoverCloseHandler = (e: MouseEvent) =>
+        e.target !== EventPassContainerRef.current && setShow?.(false);
+
+    // On clicking anywhere other than the wrapped component the pop-over closes
+    useEffect(() => {
+        document.body.addEventListener('click', onPopoverCloseHandler);
+
+        return () => {
+            document.body.removeEventListener('click', onPopoverCloseHandler);
+        };
+    }, []);
```

Comment:
you can add this hook and use it here so that in the future if we need to handle this anywhere else we can just re-use the hook
https://github.com/Brightscout/mattermost-plugin-azure-devops/blob/master/webapp/src/hooks/useClickOutside.ts